### ISSUE TITLE
LAST pipeline: clutter in log file:

### ISSUE THREAD

**Issue by EranOfek** [EranOfek/AstroPack#537]
A lot of messages in log files - hard to follow:
Instead when file is not written, just give a summary of how many images were not written.

[334338] 2024-11-27 09:57:19.773 > [INF] last02w: pipeline.DemonLAST finished executing pipeline for group 2 - start saving data / RunTime: 362.7
[334338] 2024-11-27 09:57:25.148 > [INF] last02w: pipeline.DemonLAST finished saving PROC products group 2 / RunTime: 368.1
[334338] 2024-11-27 09:57:25.979 > [INF] last02w: FileName=/last02w/data1/archive/LAST.01.02.03//2024/11/02/proc/023222v0/LAST.01.02.03_20241103.023212.430_clear_1255_000_001__sci_coadd_Image_1.fits, DataProperty=Image, image is empty - not saved
[334338] 2024-11-27 09:57:25.986 > [INF] last02w: FileName=/last02w/data1/archive/LAST.01.02.03//2024/11/02/proc/023222v0/LAST.01.02.03_20241103.023212.430_clear_1255_000_001__sci_coadd_Image_1.fits, DataProperty=Image, image is empty - not saved
[334338] 2024-11-27 09:57:25.989 > [INF] last02w: FileName=/last02w/data1/archive/LAST.01.02.03//2024/11/02/proc/023222v0/LAST.01.02.03_20241103.023212.430_clear_1255_000_001__sci_coadd_Image_1.fits, DataProperty=Image, image is empty - not saved
[334338] 2024-11-27 09:57:25.991 > [INF] last02w: FileName=/last02w/data1/archive/LAST.01.02.03//2024/11/02/proc/023222v0/LAST.01.02.03_20241103.023212.430_clear_1255_000_001__sci_coadd_Mask_1.fits, DataProperty=Mask, image is empty - not saved
[334338] 2024-11-27 09:57:25.993 > [INF] last02w: FileName=/last02w/data1/archive/LAST.01.02.03//2024/11/02/proc/023222v0/LAST.01.02.03_20241103.023212.430_clear_1255_000_001__sci_coadd_Mask_1.fits, DataProperty=Mask, image is empty - not saved
[334338] 2024-11-27 09:57:25.995 > [INF] last02w: FileName=/last02w/data1/archive/LAST.01.02.03//2024/11/02/proc/023222v0/LAST.01.02.03_20241103.023212.430_clear_1255_000_001__sci_coadd_Mask_1.fits, DataProperty=Mask, image is empty - not saved
[334338] 2024-11-27 09:57:25.998 > [INF] last02w: FileName=/last02w/data1/archive/LAST.01.02.03//2024/11/02/proc/023222v0/LAST.01.02.03_20241103.023212.430_clear_1255_000_001__sci_coadd_Cat_1.fits, DataProperty=Cat, image is empty - not saved
[334338] 2024-11-27 09:57:26.000 > [INF] last02w: FileName=/last02w/data1/archive/LAST.01.02.03//2024/11/02/proc/023222v0/LAST.01.02.03_20241103.023212.430_clear_1255_000_001__sci_coadd_Cat_1.fits, DataProperty=Cat, image is empty - not saved
[334338] 2024-11-27 09:57:26.002 > [INF] last02w: FileName=/last02w/data1/archive/LAST.01.02.03//2024/11/02/proc/023222v0/LAST.01.02.03_20241103.023212.430_clear_1255_000_001__sci_coadd_Cat_1.fits, DataProperty=Cat, image is empty - not saved
[334338] 2024-11-27 09:57:26.004 > [INF] last02w: FileName=/last02w/data1/archive/LAST.01.02.03//2024/11/02/proc/023222v0/LAST.01.02.03_20241103.023212.430_clear_1255_000_001__sci_coadd_PSF_1.fits, DataProperty=PSF, image is empty - not saved
[334338] 2024-11-27 09:57:26.006 > [INF] last02w: FileName=/last02w/data1/archive/LAST.01.02.03//2024/11/02/proc/023222v0/LAST.01.02.03_20241103.023212.430_clear_1255_000_001__sci_coadd_PSF_1.fits, DataProperty=PSF, image is empty - not saved
[334338] 2024-11-27 09:57:26.008 > [INF] last02w: FileName=/last02w/data1/archive/LAST.01.02.03//2024/11/02/proc/023222v0/LAST.01.02.03_20241103.023212.430_clear_1255_000_001__sci_coadd_PSF_1.fits, DataProperty=PSF, image is empty - not saved
[334338] 2024-11-27 09:57:26.011 > [INF] last02w: pipeline.DemonLAST finished saving COADD products group 2 / RunTime: 368.9
[334338] 2024-11-27 09:57:27.627 > [INF] last02w: FileName=/last02w/data1/archive/LAST.01.02.03//2024/11/02/proc/023222v0/LAST.01.02.03_NaN_clear_1255_000_001_018_sci_merged_Cat_1.fits, DataProperty=CatData, image is empty - not saved
[334338] 2024-11-27 09:57:27.630 > [INF] last02w: FileName=/last02w/data1/archive/LAST.01.02.03//2024/11/02/proc/023222v0/LAST.01.02.03_NaN_clear_1255_000_001_023_sci_merged_Cat_1.fits, DataProperty=CatData, image is empty - not saved
[334338] 2024-11-27 09:57:27.633 > [INF] last02w: FileName=/last02w/data1/archive/LAST.01.02.03//2024/11/02/proc/023222v0/LAST.01.02.03_NaN_clear_1255_000_001_024_sci_merged_Cat_1.fits, DataProperty=CatData, image is empty - not saved
[334338] 2024-11-27 09:57:29.849 > [INF] last02w: FileName=/last02w/data1/archive/LAST.01.02.03//2024/11/02/proc/023222v0/LAST.01.02.03_NaN_clear_1255_000_001_018_sci_merged_MergedMat_1.hdf5, DataProperty=MergedMat, image is empty - not saved

**Comment by agioffe** [comment]
OK, I will deal with it.

**Comment by agioffe** [comment]
I added a couple of lines to **pipeline.DemonLAST.writeLog** so now instead of listing each of the empty images that were not saved as disk file it just writes 1 line to say that a certain number of empty images have been skipped. Tested it on one LAST visit where I deleted some of the AllSI images by hand just prior to a call to **imProc.io.writeProduct**. This is how it now looks in the log:

16:50:18.385 [INF] WRX80: procMergeCoadd: coadding field 21
16:50:21.276 [INF] WRX80: procMergeCoadd: coadding field 22
16:50:24.164 [INF] WRX80: procMergeCoadd: coadding field 23
16:50:27.011 [INF] WRX80: procMergeCoadd: coadding field 24
16:50:46.283 [INF] WRX80: pipeline.DemonLAST finished executing pipeline for group 1 - start saving data / RunTime: 348.8
16:51:54.158 [INF] WRX80: 16 empty images were not written
16:51:54.165 [INF] WRX80: pipeline.DemonLAST finished saving PROC products group 1 / RunTime: 416.6
16:51:55.826 [INF] WRX80: pipeline.DemonLAST finished saving COADD products group 1 / RunTime: 418.3

**Comment by EranOfek** [comment]
But writeLog is a low level function.
I would think that this should be fone by the high level function?!

**Comment by agioffe** [comment]
I would say that **pipeline.DemonLAST.writeLog** is a medium level function, because it is between the real low-level function Obj.**Logger.msgLog** and the higher-level **imProc.io.writeProduct**. If you consider it reasonable, I can move this functionality 1 step upward, to **imProc.io.writeProduct** (moving further up to **pipeline.DemonLAST.main** is definitely not reasonable).

**Comment by EranOfek** [comment]
Not sure.
Lets discuss this tomorrow
